python/tskit/trees.py

```diff
@@ -5535,15 +5535,30 @@ def genetic_relatedness(
             segregating sites (defaults to True).
         :return: A ndarray with shape equal to (num windows, num statistics).
         """
-        return self.__k_way_sample_set_stat(
-            self._ll_tree_sequence.genetic_relatedness,
-            2,
-            sample_sets,
-            indexes=indexes,
-            windows=windows,
-            mode=mode,
-            span_normalise=span_normalise,
-            polarised=polarised,
+        if proportion:
+            # TODO this should be done in C also
+            all_samples = list({u for s in sample_sets for u in s})
```

Comment:
Why do we need to do this ``all_samples`` step? I would have thought we could pass the ``sample_sets`` through without changes.

Reply:
So, if `sample_sets=[[u] for u in ts.samples()]` then `ts.segregating_sites(sample_sets)` is applied to each sample independently (so returns a list of zeros of length num_samples).

Now... I can't remember why I was writing `sample_sets` in this way but must have copy-pasted it from somewhere else!

Reply:
I agree: we want the denominator to be the number of sites that are segregating in *any* of the samples of any of the sample sets. (need to make sure to document that it is this, not segregating in all samples of the trees sequence)

Reply:
Just to clarify - this means I should leave this bit as is?

Reply:
Yes, that's my interpretation of @petrelharp's comment @brieuclehmann. Looks like we should update the documentation to clarify the point, though

Reply:
Yep, I'll update that on the docs PR.